### PR TITLE
feat(composer): add a11y color picker

### PR DESCRIPTION
**Pull request by jacksonvogt**
## Overview
New, accessible color picker built on top of [react-colorful](https://omgovich.github.io/react-colorful/).

Major changes from first revision
- Got rid of save/cancel flow upon discussions with @divya-sea and Brenton (auto saving now)
- Refactored sub-components of color picker
- Got rid of styled-components
- Got rid of managing color types, using ColorRepresentation everywhere now
- Use tinycolor for color conversions, validation, etc.
- Renamed component to ColorPicker from CustomColorPicker
- Mocked polaris in tests, having issues with fireEvent on the mocked components though so have low test coverage for the moment

Demo video:

https://github.com/awslabs/iot-app-kit/assets/143754227/610def65-8744-49b1-946c-7ca01b24e97a

### Scene Composer
For `scene-composer` package changes specifically, you can preview the component in the published storybook artifact. To do this, wait for the `Publish Storybook` action to complete below.

- Click on the workflow details
- Select the Summary item on the left
- Download the zip file

To run the storybook build locally, you need a local static web server:

```
npm install -g httpserver
cd <Extracted Zip Directory>
httpserver
```

Then open the website http://localhost:8080 to run the doc site.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/scene-composer/src/components/panels/ColorPicker/ColorPicker.scss [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+.color-picker .react-colorful__saturation {
+  border-radius: 5px;
+  border-bottom: none;
+}
+
+.color-picker .react-colorful__hue {
+  margin-top: 24px;
+  margin-bottom: 20px;
```
Comment (mumanity) [inline packages/scene-composer/src/components/panels/ColorPicker/ColorPicker.scss:14]: Things that are assigning visual sizes such as margins, height, width, & padding should use `rem` or `em` in order to maintain responsive & consistent appearance between browsers & monitors. 

Things that control edges (borders, radius, etc...) are where use of `px` are appropriate. 

We have not standardized on whether we're going to be using `rem` or `em`, but would strongly encourage implementing either.

File: packages/scene-composer/src/components/panels/ColorPicker/ColorPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      <form onSubmit={(e) => e.preventDefault()}>
+        <ReactColorfulPicker color={hexString(internalColor)} onChange={debounce(handleColorfulChange, 100)} />
+        <FlexibleDiv>
+          <HexInputField
+            color={`#${hexInputValue.replace('#', '')}`}
+            onChange={(e) => handleHexChange(e.detail.value)}
+            label={formatMessage({ defaultMessage: 'Hex', description: 'Abbreviation for hexidecimal' })}
+            errorText={
+              !isValidColor(hexInputValue) &&
+              formatMessage({ defaultMessage: 'Invalid Hex', description: 'Form field error message' })
```
Comment (mumanity) [inline packages/scene-composer/src/components/panels/ColorPicker/ColorPicker.tsx:64]: Please implement `react-intl` for strings that appear in the UI. 

[Example here. ](https://github.com/awslabs/iot-app-kit/blob/8eee582e039a5cb2a3e25c81f822a4ecca95672c/packages/scene-composer/src/components/panels/scene-components/AnchorComponentEditor.tsx#L197)

File: packages/scene-composer/src/components/panels/ColorPicker/ColorPickerHelpers.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  return `#${value.replace(/[^a-fA-F0-9]/g, '').substring(0, 6)}`;
+};
+
+export const getRgb = (color: ColorRepresentation) => {
+  return toTinyColor(color).toRgb();
+};
+
+export const validateRgbValue = (value: string) => {
+  const onlyDigits = value.replace(/\D/g, '');
+  return parseInt(onlyDigits) > 255 ? '255' : parseInt(onlyDigits.padStart(1, '0')).toString();
```
Comment (mumanity) [inline packages/scene-composer/src/components/panels/ColorPicker/ColorPickerHelpers.tsx:36]: Curious about 255 as a set cap here. Might be worth assigning to a `const` with a more meaningful name.

File: packages/scene-composer/src/components/panels/ColorPicker/ColorPickerHelpers.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      <Input {...props} />
+    </div>
+  );
+};
+
+export const HexInputField = ({ color, onChange, label, errorText }) => {
+  return (
+    <FormField label={label} errorText={errorText}>
+      <FixedWidthInput
+        width='75px'
```
Comment (mumanity) [inline packages/scene-composer/src/components/panels/ColorPicker/ColorPickerHelpers.tsx:71]: Same comment as above. Using `px` for set widths is an inconsistent user experience as it is not properly responsive.

File: packages/scene-composer/src/components/panels/ColorPicker/ColorPicker.scss (hunk 1 above)
Comment (jacksonvogt) [reply packages/scene-composer/src/components/panels/ColorPicker/ColorPicker.scss]: Fixed

Comment (jacksonvogt) [reply packages/scene-composer/src/components/panels/ColorPicker/ColorPicker.scss]: Good point! Fixed

File: packages/scene-composer/src/components/panels/ColorPicker/ColorPickerHelpers.tsx (hunk 3 above)
Comment (jacksonvogt) [reply packages/scene-composer/src/components/panels/ColorPicker/ColorPickerHelpers.tsx]: Fixed

File: packages/scene-composer/src/components/panels/ColorPicker/ColorPicker.scss (hunk 1 above)
Comment (divya-sea) [reply packages/scene-composer/src/components/panels/ColorPicker/ColorPicker.scss]: Good point @mumanity

File: packages/scene-composer/package.json [hunk 5]
```diff
@@ -161,6 +161,7 @@
     "path-browserify": "1.0.1",
```
Comment (mumanity) [inline packages/scene-composer/package.json:161]: This CR is showing 7 commits, however, our policy is that only one commit goes in per review. To resolve this, please do the following from your local machine:

* git reset HEAD~7
* add all files to tracking (ex: git add *)
* git commit -m 'YOUR_MESSAGE_HERE`
* git push --force

**Review by mumanity — CHANGES_REQUESTED** [review]

File: packages/scene-composer/package.json (hunk 5 above)
Comment (jacksonvogt) [reply packages/scene-composer/package.json:161]: Saw this too late, tried an interactive rebase and accidentally included all the commits that I tried to resolve merge conflicts with earlier🤦‍♂️. Trying to fix now

Comment (jacksonvogt) [reply packages/scene-composer/package.json:161]: Fixed :)

**Review by divya-sea — APPROVED** [review]

File: packages/scene-composer/src/components/panels/scene-rule-components/SceneRuleTargetEditor.tsx [hunk 6]
```diff
@@ -98,7 +98,7 @@ export const SceneRuleTargetEditor: React.FC<ISceneRuleTargetEditorProps> = ({
             chosenColor={chosenColor}
           />
           {isCustomStyle && (
-            <ColorPicker
+            <TagColorPicker
```
Comment (divya-sea) [inline packages/scene-composer/src/components/panels/scene-rule-components/SceneRuleTargetEditor.tsx:101]: We have used this component in other components as well. So keep a generic name as well something like `ColorSelectorCombo`

**Review by mumanity — APPROVED** [review]

**Pull request merged**